sap-service-operator/src/main/java/com/sap/cloud/environment/servicebinding/DirectoryBasedCache.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved.
+ */
+
+package com.sap.cloud.environment.servicebinding;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.annotation.Nonnull;
+
+import com.sap.cloud.environment.servicebinding.api.ServiceBinding;
+
+@FunctionalInterface
+interface DirectoryBasedCache
+{
+    @Nonnull
+    List<ServiceBinding> getServiceBindings( @Nonnull final Collection<Path> directories );
+
+    @Nonnull
+    default List<ServiceBinding> getServiceBindings( @Nonnull final Stream<Path> directories )
```

Comment:
(Question)
General question out of interest, why would you add `Stream` as argument for API?

Reply:
Because it is quite convenient to have this API as it is commonly used in combination with `Files.list()`, which returns a `Stream<Path>`.